dotcom-rendering/src/components/CartoonComponent.tsx

```diff
@@ -68,14 +68,14 @@ export const CartoonComponent = ({ element }: Props) => {
 			</Hide>
 
 			<Caption
-				captionText={element.caption}
+				{...(!!element.caption && { captionText: element.caption })}
 				format={{
 					display: ArticleDisplay.Standard,
 					design: ArticleDesign.Standard,
 					theme: Pillar.News,
 				}}
-				credit={element.credit}
-				displayCredit={element.displayCredit}
+				{...(!!element.credit && { credit: element.credit })}
+				displayCredit={element.displayCredit ?? true}
```

Comment:
> this means that if the `displayCredit` field is missing the default behaviour should be to show the credit?

I don’t think it could ever be missing? Unless in old data? If this is the concern, then I suppose yes: it should show.

The usual behaviour of these things is: when `displayCredit` is `false` **and** `caption` is empty, the whole component (caption+credit) should not render.

Reply:
ignore this... it was an attempt to fix linting. As the IDE was complaining about passing in undefined values, I assumed it was the culprit, but it was actually unhappy with something else.

On `displayCredit` (and other fields), the content-api-model library that frontend consumes assumes almost all fields are optional, even if they are compulsory upstream. I guess this forces consuming applications to decide a) if they need the field in order to render the element and b) if there is a sensible default to apply.

DCR currently applies `displayCredit=true` in the event the displayCredit field is missing in the image element, so we've continued this approach with cartoons.